Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1178,7 +1178,7 @@ class BrowserViewController: UIViewController {
             //Disable any sort of privileged execution contexts
             //IE: The user must explicitly type OR must explicitly tap a bookmark they have saved.
             //Block all other contexts such as redirects, downloads, embed, linked, etc..
-            if visitType == .typed || visitType == .bookmark {
+            if visitType == .bookmark {
```

Comment:
Looks good to me. This will explicitly block the user from typing in any javascript and `copy-paste`.